core/src/main/java/com/yahoo/oak/HashChunk.java

```diff
@@ -200,11 +199,16 @@ int compareKeyAndEntryIndex(KeyBuffer tempKeyBuff, K key, int ei) {
      *                   This means that there is an entry with that key, but there is no value attached to this key.
      *                   Such entry can be reused after finishing the deletion process, if needed.
      * @param key the key to look up
+     * @param forGetsOnly
      */
-    void lookUp(ThreadContext ctx, K key) {
+    void lookUp(ThreadContext ctx, K key, boolean forGetsOnly) {
         int keyHash = calculateKeyHash(key, ctx);
         int idx = calculateEntryIdx(key, keyHash);
-        entryHashSet.lookUp(ctx, key, idx, keyHash);
+        if (forGetsOnly) {
+            entryHashSet.lookUpForGetOnly(ctx, key, idx, keyHash);
+        } else {
+            entryHashSet.lookUp(ctx, key, idx, keyHash);
+        }
```

Comment:
IMO, it is better to add another function: `HashChunk.lookUpForGetOnly()` rather then add a boolean parameter.
This way, you'll only have to change the places that actually use this method, instead of all the calls to `lookup()`.

Reply:
Good idea, although `HashChunk.lookUpForGetOnly()` and `HashChunk.lookUp()` are going to be very similar... But fixed anyway